go/v1/resource_descriptor.go

```diff
@@ -5,29 +5,51 @@ Wrapper APIs for in-toto attestation ResourceDescriptor protos.
 package v1
 
 import (
+	"crypto/md5"
+	"crypto/sha1"
+	"crypto/sha512"
 	"encoding/hex"
 	"errors"
 )
 
 var (
+	ErrIncorrectDigestLength = errors.New("digest is not correct length")
 	ErrInvalidDigestEncoding = errors.New("digest is not valid hex-encoded string")
 	ErrRDRequiredField       = errors.New("at least one of name, URI, or digest are required")
 )
 
+// Supported standard hash algorithms
+func isSupportedAlgorithm(alg string) (bool, int) {
+	algos := map[string]int{"md5": md5.Size, "sha1": sha1.Size, "shake128": md5.Size, "sha224": sha512.Size224, "sha3_224": sha512.Size224, "sha512_224": sha512.Size224, "sha256": sha512.Size256, "sha3_256": sha512.Size256, "sha512_256": sha512.Size256, "shake256": sha512.Size256, "sha384": sha512.Size384, "sha3_384": sha512.Size384, "sha512_384": sha512.Size384, "sha512": sha512.Size, "sha3_512": sha512.Size, "dirHash": sha512.Size256, "gitCommit": sha1.Size}
+
+	size, ok := algos[alg]
+	return ok, size
+}
+
 func (d *ResourceDescriptor) Validate() error {
 	// at least one of name, URI or digest are required
 	if d.GetName() == "" && d.GetUri() == "" && len(d.GetDigest()) == 0 {
 		return ErrRDRequiredField
 	}
 
-	// the in-toto spec expects a hex-encoded string in DigestSets
-	// https://github.com/in-toto/attestation/blob/main/spec/v1/digest_set.md
 	if len(d.GetDigest()) > 0 {
-		for _, digest := range d.GetDigest() {
-			_, err := hex.DecodeString(digest)
+		for alg, digest := range d.GetDigest() {
+
+			// check encoding and length for supported algorithms
+			supported, size := isSupportedAlgorithm(alg)
```

Comment:
Can you add a comment like "Users are allowed to use unsupported algorithms and their use should not generate a validation error."